pkg/controller/controller.go

```diff
@@ -84,6 +86,25 @@ func NewRestConfig(conf *config.Config) *rest.Config {
 }
 
 func NewManagerForRestConfig(conf *config.Config, rc *rest.Config) (ctrl.Manager, error) {
+	// client creates ListWatch cache for all resources that are
+	// GET or LISTed. this uses too much memory on large clusters.
+	var clientOpts *client.CacheOptions
+	cacheOpts := cache.Options{}
+	if conf.DisableExpensiveCache {
+		clientOpts = &client.CacheOptions{
+			// we only interact with a small subset of pods
+			// but there's no good filter to only cache the pods we care about
```

Comment:
why not top-level filters for pods? don't we apply them to everything we deploy?

Reply:
the placeholder pod doesn't actually have top level labels (it's the deployment that does)

https://github.com/Azure/aks-app-routing-operator/blob/175e5e1fe8b429f9c49254a37be4ec53f0c60785/pkg/controller/keyvault/placeholder_pod.go#L284

Good idea for an improvement here though. I'll add that in a future iteration